src/rebar3_format_prv.erl

```diff
@@ -103,27 +103,29 @@ get_output_dir(Action, Args) ->
           none
     end.
 
--spec get_formatter(rebar_state:t()) -> module().
-get_formatter(State) ->
-    proplists:get_value(formatter, rebar_state:get(State, format, []), default_formatter).
+-spec get_formatter(rebar_state:t(), rebar3_formatter:opts()) -> rebar3_formatter:t().
+get_formatter(State, Opts) ->
+    Module = proplists:get_value(formatter,
+                                 rebar_state:get(State, format, []),
+                                 default_formatter),
+    rebar3_formatter:new(Module, Opts).
 
 -spec get_opts(rebar_state:t()) -> rebar3_formatter:opts().
 get_opts(State) ->
     proplists:get_value(options, rebar_state:get(State, format, []), #{}).
 
--spec format([file:filename_all()], module(), rebar3_formatter:opts()) -> ok |
-                                                                          {error, term()}.
-format(Files, Formatter, Opts) ->
+-spec format_file([file:filename_all()], rebar3_formatter:t()) -> ok | {error, term()}.
```

Comment:
maybe it should be called `format_files` or something like that because it takes a list of files as parameter instead of a single file.